### ISSUE TITLE
@parcel/validator-typescript breaks with typescript 4.4.2

### ISSUE THREAD

**Issue by astegmaier** [parcel-bundler/parcel#6779]
# 🐛 bug report

`@parcel/validator-typescript` will fail in these conditions:
- your project uses typescript `4.4.2` ([released August 23, 2021](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/))
- you have a custom `tsconfig.json` file in your project (probably due to #4138)

## 🎛 Configuration (.babelrc, package.json, cli command)

***package.json***
```json
{
  "scripts": {
    "build": "parcel build src/index.html",
  },
  "dependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@parcel/validator-typescript": "2.0.0-rc.0",
    "parcel": "2.0.0-rc.0",
    "typescript": "4.4.2"
  }
}
```

***.parcelrc***
```json
{
    "extends": "@parcel/config-default",
    "validators": {
        "*.{ts,tsx}": [
            "@parcel/validator-typescript",
        ]
    },
}
```

***tsconfig.json***
```
<any valid configuration>
```

## 🤔 Expected Behavior

Parcel should build successfully

## 😯 Current Behavior

`parcel build` fails with this error:

```
yarn run v1.22.11
$ parcel build src/index.html
🚨 Build failed.

@parcel/validator-typescript: directoryExists is not a function

  TypeError: directoryExists is not a function
  at Object.matchFiles (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/typescript/lib/typescript.js:20102:17)
  at ParseConfigHost.readDirectory (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/@parcel/ts-utils/lib/FSHost.js:105:20)
  at getFileNamesFromConfigSpecs (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/typescript/lib/typescript.js:40747:40)
  at getFileNames (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/typescript/lib/typescript.js:40250:29)
  at parseJsonConfigFileContentWorker
  (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/typescript/lib/typescript.js:40183:24)
  at Object.parseJsonConfigFileContent
  (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/typescript/lib/typescript.js:40124:16)
  at tryCreateLanguageService
  (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/@parcel/validator-typescript/lib/TypeScriptValidator.js:138:53)
  at /Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/@parcel/validator-typescript/lib/TypeScriptValidator.js:87:7
  at async Promise.all (index 0)
  at async Object.validateAll
  (/Users/Andrew/Projects/parcel-typescript-validator-broken/node_modules/@parcel/validator-typescript/lib/TypeScriptValidator.js:81:5)
error Command failed with exit code 1.
```

## 💁 Possible Solution

It looks like there is code in parcel's ts-utils that is responsible for looking for custom `tsconfig.json` files, and it ultimately makes a call to typescript's `matchFiles` api:

https://github.com/parcel-bundler/parcel/blob/feb67dbf9d51ad567ca54597d4f50d5808a164ed/packages/utils/ts-utils/src/FSHost.js#L93-L104

It seems that the signature of the `matchFiles` function has changed in `4.4.2` so that there is a new required parameter at the end called `directoryExists` - see [this PR](https://github.com/microsoft/TypeScript/pull/44710/files#diff-1516c8349f7a625a2e4a2aa60f6bbe84e4b1a499128e8705d3087d893e01d367)

It looks like we already have a method called `directoryExists` on the `FSHost` class:

https://github.com/parcel-bundler/parcel/blob/feb67dbf9d51ad567ca54597d4f50d5808a164ed/packages/utils/ts-utils/src/FSHost.js#L40

So the fix might be as simple as passing a function like this as the last parameter of the call to `matchFiles`
```js
dirPath => this.directoryExists(dirPath)
```

## 🔦 Context

I want to use the latest version of typescript in my projects, along with `@parcel/validator-typescript`.

## 💻 Code Sample

You can see a repro of this bug [in this repo](https://github.com/astegmaier/parcel-typescript-validator-broken).

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-rc.0
| Node             | 14.17.5
| Yarn         | 1.22.11
| Operating System | Mac OS X 11.5.2

**Comment by alexreardon** [comment]
I ran into this as well. Which version of parcel should I use to get this fix? I'm currently on `2.0.0-rc.0`

**Comment by mischnic** [comment]
`yarn add @parcel/validator-typescript@nightly`

**Comment by Luxcium** [comment]
What is the next stable release to fix the issue **or** when will it be awailable?

![Screenshot_20210903_070229](https://user-images.githubusercontent.com/42672814/131995453-f600d922-b2d2-4e00-aeec-4be6c4e42bd1.png)

>I ran into this as well. Which version of parcel should I use to get this fix? I'm currently on `2.0.0-rc.0`
>_Originally posted by @alexreardon in https://github.com/parcel-bundler/parcel/issues/6779#issuecomment-908940742_

using nightly fixed the issue:
```
{
  "devDependencies": {
    "@parcel/transformer-typescript-tsc": "^2.0.0-nightly.826",
    "@parcel/transformer-typescript-types": "^2.0.0-nightly.826",
    "parcel": "^2.0.0-nightly.824",
    "typescript": "~4.4.2",
    "tslib": "latest",
    "@types/node": "^16.7.4"
  }
}

```
